py/objstr.c

```diff
@@ -2098,11 +2098,11 @@ bool mp_obj_str_equal(mp_obj_t s1, mp_obj_t s2) {
 
 STATIC NORETURN void bad_implicit_conversion(mp_obj_t self_in) {
     if (MICROPY_ERROR_REPORTING == MICROPY_ERROR_REPORTING_TERSE) {
-        mp_raise_TypeError("can't convert to str implicitly");
+        mp_raise_TypeError("Can't convert to str implicitly");
```

Comment:
Most (I thought it was "all" but apparently not) of the CPython error messages start with a lower case letter, so that's what we try and do consistently in uPy.  And we don't match 100% with CPython error messages because it's not really important.  So I don't see a reason to make this change.

Reply:
The reason I changed is that the test would otherwise fail (apparently run-tests is testing against CPython?). But with the changes suggested below this isn't necessary anymore.